File: Back-end/Coffee-Shop/src/main/java/vn/com/ltdt/Coffee_Shop/product/dtos/ProductDetailDTO.java
package vn.com.ltdt.Coffee_Shop.product.dtos;

import jakarta.validation.constraints.NotNull;

public record ProductDetailDTO(
        int id,
        @NotNull(message = "Product ID cannot be null")
        int productId,
        String size,
        @NotNull(message = "Product Price cannot be null")
        double price
) {

}

File: Font-end/Mobile/AndroidNative/app/src/main/java/com/ltdt/coffeeshop_android_native/data/remote/ProductDetailDTO.kt
package com.ltdt.coffeeshop_android_native.data.remote

import com.ltdt.coffeeshop_android_native.data.domains.ProductDetail

data class ProductDetailDTO(
    val id: Int,
    val productId: Int,
    val size: String,
    val price: Double
)

fun ProductDetailDTO.toEntity() = ProductDetail(
    id = id,
    productId = productId,
    size = size,
    price = price
)